utils/utils.go

```diff
@@ -68,7 +82,9 @@ func WalkFiles(path string, fn func(path string, info os.FileInfo)) {
 		if !NormalFile(info) { // **THIS IS WHAT SKIPS DIRECTORIES**
 			return nil
 		}
-		filesCh <- PathAndInfo{path, info}
+		if !ignoreErrors || HaveReadPermission(path) {
```

Comment:
I think this would make more sense like this:

```
if ignoreErrors && !HaveReadPermission(path) {
    return nil // skip
}
filesCh <- PathAndInfo{path, info}
```